shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/dml/InsertStatement.java

```diff
@@ -43,7 +43,7 @@
  */
 @Getter
 @Setter
-public final class InsertStatement extends AbstractSQLStatement implements DMLStatement {
+public abstract class InsertStatement extends AbstractSQLStatement {
```

Comment:
Since some segments are just for the particular database, it is required to review and split these segments into its child classes, IMO. Like `WithSegment` is supposed to appear in `SQLServerInsertStatement`, right?

Reply:
Hi @tristaZero , yes, we should split these segments into child classes, I want to finish this work in another PR, because in one PR, it will be modify too much files and make the code review difficultly.

and I will communicate with jingshang right now, make sure the work not conflict.

Reply:
Thanks for your explanation. @jingshanglu Could you give it a look to check whether there will be any conflicts?